### ISSUE TITLE
discoverDomain is missing JMS resource details

### ISSUE THREAD

**Issue by zeusbaba** [oracle/weblogic-deploy-tooling#792]
I did export WDT template using `discoverDomain`, from `12.1.3.0.0`, it worked without issues.  
However, after analysing the template, I noticed that the details of JMS resource was missing.  

the original resource file contains  
```
<foreign-server name="UMSAQJMSForeignServer">
    <default-targeting-enabled>true</default-targeting-enabled>
    <foreign-destination name="OraSDPM/Queues/OraSDPMAppDefRcvQ1">
      <local-jndi-name>OraSDPM/Queues/OraSDPMAppDefRcvQ1</local-jndi-name>
      <remote-jndi-name>Queues/OraSDPMAppDefRcvQ1</remote-jndi-name>
    </foreign-destination>
    ...

```  

but the exported WDT template is missing `foreign-destination / ForeignDestination` details  
```
ForeignDestination:
    OraSDPM/Queues/OraSDPMAppDefRcvQ1:
    OraSDPM/Queues/OraSDPMAppDefRcvErrorQ1:
    ...
```  
as far as I know, the expected output should be   
```
OraSDPM/Queues/OraSDPMAppDefRcvQ1:
    LocalJNDIName: OraSDPM/Queues/OraSDPMAppDefRcvQ1
    RemoteJndiName: Queues/OraSDPMAppDefRcvQ1
...
```

**Comment by CarolynRountree** [comment]
Could you verify the version of  WDT please? At the bottom of the stdout, there should be a version number. Or in the top directory of the install is a VERSION.txt file.

**Comment by zeusbaba** [comment]
i'm using the version from 18nov2020, I did create a package from `master` branch incl [this commit](https://github.com/oracle/weblogic-deploy-tooling/commit/3bf40be38603c32d893a6f5b63c8657e0a6cb3a1)

**Comment by CarolynRountree** [comment]
I am unable to recreate the problem with my own foreign server. Could you provide the entire foreign server config entry so I can drop it into my domain and see how it works.

**Comment by CarolynRountree** [comment]
Looks like this is related to the slashes in the names and attribute values with WLS 12.1.3. We will see if we can work around this problem in WDT.

**Comment by zeusbaba** [comment]
in my original issue, I included `<foreign-server name="UMSAQJMSForeignServer">...` entry as an example as well.  
is is sufficient? or do you need the whole file?

**Comment by rakillen** [comment]
We have a model to recreate the problem now, so no need to post yours. Looking into a solution for the slash names in 12.1.3.

**Comment by ddsharpe** [comment]
Unfortunately, WLST 12.1.3 (WebLogic Scripting Tool) has known issues with the slash (/) in object names.  We looked into coding around these issues in WDT but the workaround that is required would only apply to WLS 12.1.x because the workaround breaks in 12.2.1.x (the next version of WebLogic Server).  And, the workaround only worked for discovering the domain, and creating domains with slashes in names was still problematic.  We, also, found that the WebLogic documentation for 12.1.3 recommends that slashes not be used in object names at all.  

We do not have plans to implement a workaround for slashes in the object names for WLS 12.1.x.  But, the slashes in the names for version 12.2.1.x domains should work.

Do you need the discovered model to create a 12.1.3 domain?  Or were you planning on upgrading to a newer version of WebLogic?

**Comment by zeusbaba** [comment]
thnx for your efforts and feedback.  
the current need is to use WDT for upgrading to 12.2.1.x, and  I can confirm that it works fine.  
using WDT for creating 12.1.3 domain is of a lower priority in this period.  

However, notice that this issue #792 is related to `discoverDomain` from already set up 12.1.3. 
I did introspect an already set up 12.1.3 so to produce WDT template, afterwards I did many manual fixes to that template so that to use for `createDomain` of 12.2.1.x domain. 
one of the manual fixes was about the empty `ForeignDestination:` values.  

I can share as a feedback about WDT, the expected flow is as below: 
- `discoverDomain` should produce a template by introspecting an already set up domain   
- the produced WDT template should be usable (with minimal intervention) to create a new domain   
both steps should be version agnostic.  
it _might_ sound like a far stretched goal, but it brings a high value for teams who need to frequently create domains of diferent versions (both 12.1.3 and 12.2.1.x)
